kernel/src/sched/stats/loadavg.rs

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module implements the CPU load average calculation.
+//!
+//! Reference: <https://github.com/torvalds/linux/blob/master/kernel/sched/loadavg.c>
+
+use lazy_static::lazy_static;
+use ostd::{arch::timer::TIMER_FREQ, sync::RwLock, timer};
+
+/// nr of bits of precision
+pub const FSHIFT: u64 = 11;
+/// 1.0 as fixed-point
+pub const FIXED_1: u64 = 1 << FSHIFT;
+/// 5 sec intervals
+pub const LOAD_FREQ: u64 = 5 * TIMER_FREQ + 1;
+/// 1/exp(5sec/1min) as fixed-point
+pub const EXP_1: u64 = 1884;
+/// 1/exp(5sec/5min)
+pub const EXP_5: u64 = 2014;
+/// 1/exp(5sec/15min)
+pub const EXP_15: u64 = 2037;
+
+/// Load average of all CPU cores.
+///
+/// The load average is calculated as an exponential moving average of the load
+/// over the last 1, 5, and 15 minutes.
+#[derive(Default, Clone)]
+pub struct LoadAverage {
+    load: [u64; 3],
+    last_update: u64, // jiffies
+}
+
+lazy_static! {
+    /// The global load average.
+    pub static ref LOAD_AVG: RwLock<LoadAverage> = RwLock::new(LoadAverage::default());
+}
```

Comment:
The general practice is not to use `lazy_static`, as it's both discouraged by the Rust community and have problems in our system due to the difference in the implementation of the synchronization.

In this case, `LoadAverage` can clearly have a `const new()` function that can serve static initialization.

Reply:
It can indeed, good catch